src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -182,48 +187,78 @@ private boolean isTerminated(final DetailAST ast, boolean useBreak,
                 break;
             case TokenTypes.LITERAL_CONTINUE:
-                terminated = useContinue;
+                terminated =
+                        useContinue || hasLabel(ast, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.SLIST:
-                terminated = checkSlist(ast, useBreak, useContinue);
+                terminated =
+                        checkSlist(ast, useBreak, useContinue, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_IF:
-                terminated = checkIf(ast, useBreak, useContinue);
+                terminated =
+                        checkIf(ast, useBreak, useContinue, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_FOR:
             case TokenTypes.LITERAL_WHILE:
             case TokenTypes.LITERAL_DO:
-                terminated = checkLoop(ast);
+                terminated = checkLoop(ast, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_TRY:
-                terminated = checkTry(ast, useBreak, useContinue);
+                terminated =
+                        checkTry(ast, useBreak, useContinue, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_SWITCH:
-                terminated = checkSwitch(ast, useContinue);
+                terminated =
+                        checkSwitch(ast, useContinue, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_SYNCHRONIZED:
-                terminated = checkSynchronized(ast, useBreak, useContinue);
+                terminated =
+                        checkSynchronized(ast, useBreak, useContinue, labelsForCurrentSwitchScope);
+                break;
+            case TokenTypes.LABELED_STAT:
+                labelsForCurrentSwitchScope.add(ast.getFirstChild().getText());
+                terminated =
+                        isTerminated(ast.getLastChild(), useBreak, useContinue,
+                                labelsForCurrentSwitchScope);
                 break;
             default:
                 terminated = false;
         }
         return terminated;
     }
 
+    /**
+     * Checks if given break or continue ast has outer label.
+     *
+     * @param statement break or continue node
+     * @param labelsForCurrentSwitchScope the Set labels for the current scope of the switch
+     * @return true if local label used
+     */
+    private static boolean hasLabel(DetailAST statement, Set<String> labelsForCurrentSwitchScope) {
+        return Optional.ofNullable(statement)
+                .map(DetailAST::getFirstChild)
+                .filter(child -> child.getType() == TokenTypes.IDENT)
+                .map(DetailAST::getText)
+                .map(label -> !labelsForCurrentSwitchScope.contains(label))
+                .orElse(Boolean.FALSE);
```

Comment:
```suggestion
                .ifPresent();
```

is cleaner and should make checker happy

Reply:
I actually had isPresent() but the I guess one of the CI job failed, stating some issue. I can't find it. And, had to look for something else. Will try to revert it.

Reply:
I am good to suppress checker violation for this